python/pyspark/sql/utils.py

```diff
@@ -121,7 +121,10 @@ def require_minimum_pandas_version():
     from distutils.version import LooseVersion
     try:
         import pandas
+        have_pandas = True
     except ImportError:
+        have_pandas = False
+    if not have_pandas:
```

Comment:
So, this is for making the error message clean when it's reraised in Python 3? I think it's fine to leave it as is. I believe the downside of it is to lose the information where exactly the error was thrown in Python 3.

Reply:
I think having the traceback to the `raise ImportError` below is all the information needed.  If that happens, then the only possible cause is that the import failed from here.  The problem with how it was before is that for Python 3, it will print out `During handling of the above exception, another exception occurred:` which makes it seem like it is not being handled correctly, since it's really just a failed import.